### PR TITLE
Add signable representations

### PR DESCRIPTION
**Pull request by nc6**

### CODE REVIEW

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/OCert.hs [hunk 1]
```diff
@@ -129,3 +138,18 @@ kesPeriod (SlotNo s) =
...
+
+-- | Signable part of an operational certificate
+data OCertSignable crypto
+  = OCertSignable !(VerKeyKES crypto) !Natural !KESPeriod
+
+instance Crypto crypto => SignableRepresentation (OCertSignable crypto) where
+  getSignableRepresentation (OCertSignable vk counter period) =
+    KES.rawSerialiseVerKeyKES vk
+      <> BSL.toStrict (Binary.encode counter)
+      <> writeBinaryWord64 (fromIntegral $ unKESPeriod period)
```
Comment (dcoutts) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/OCert.hs]: These two should both be at-most 64bit numbers, so we should serialise them the same.

tbh, we should probably use Word64 for the cert counter not natural.

**Review by dcoutts — CHANGES_REQUESTED** [review]
looks good generally, just the OCert that's a bit squiffy

**Comment by nc6** [comment]
@dcoutts I agree that the counter should probably be bounded, but on the basis that it's not, I'm wary of serialising it as if it is, since that allows a potential attack. Since we compare the counter, one could replay the ocert counter `i` with a counter of `maxBound @Word64 + i`, and have validation pass but be able to trump the real block

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/OCert.hs [hunk 2]
```diff
@@ -145,9 +143,10 @@ data OCertSignable crypto
 
 instance Crypto crypto => SignableRepresentation (OCertSignable crypto) where
   getSignableRepresentation (OCertSignable vk counter period) =
-    KES.rawSerialiseVerKeyKES vk
-      <> BSL.toStrict (Binary.encode counter)
-      <> writeBinaryWord64 (fromIntegral $ unKESPeriod period)
+    BSL.toStrict . Binary.runPut $ do
+      Binary.putByteString (KES.rawSerialiseVerKeyKES vk)
+      Binary.put counter
+      Binary.putWord64be (fromIntegral $ unKESPeriod period)
```
Comment (dcoutts) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/OCert.hs]: The bytestring builder will be faster
```suggestion
    runByteBuilder (32+8+8) $ do
      BS.byteStringCopy (KES.rawSerialiseVerKeyKES vk)
      BS.word64BE counter
      BS.word64BE (fromIntegral $ unKESPeriod period)
```

Needs
```
import Shelley.Spec.Ledger.Serialization (runByteBuilder)
import qualified Data.ByteString.Builder as BS
import qualified Data.ByteString.Builder.Extra as BS
```

**Review by dcoutts — APPROVED** [review]
LGTM.

We should add a TODO to use a newtype wrapper for the cert counter.

**Pull request merged**